internal/middleware/id_validation.go

```diff
@@ -0,0 +1,26 @@
+package middleware
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+var (
+	idValidation = regexp.MustCompile("^[0-9]+$")
+	ErrInvalidId = fmt.Errorf("invalid id")
+)
+
+func DigitsOnly(id string) error {
+	if !idValidation.MatchString(id) {
+		return fmt.Errorf("%w: must be a greater than zero", ErrInvalidId)
+	}
+
+	// Checking for out of range error
+	_, err := strconv.ParseUint(id, 10, 64)
+	if err != nil {
+		return fmt.Errorf("%w: id too high", ErrInvalidId)
+	}
+
+	return nil
+}
```

Comment:
Do not assume that primary keys in databases are higher than 0, in fact, when a database reaches the maximum value it starts from -MAX back to zero. This is at least behavior of postgres. So I would drop this check only keeping `ParseUint` and checking for error. In addition, I would expect this function to actually error out when called with negative number anyway.

I would not bother too much about the error message, we have properly documented what is expected, just returning "invalid id" is enough.

Reply:
Now that I think about it, you should not use `ParseUint` because our primary keys are `int64`, just parse a regular 64 signed number and report errors.

Reply:
Yeah, sorry. There was even an invalid string, this regex does not check against 0, but numbers overall. The second check is against numbers that might overflow.

The second check would not fail for negative numbers. So I think we can keep both.